### ISSUE TITLE
landscape UI 

### ISSUE THREAD

**Issue by Edoardo-M** [westnordost/StreetComplete#402]
when using the app (android 0.14) in landscape mode, the answares box takes over the full space and the sub-answares are invisible until you scroll. 

A better Setup will be to have the box entering from the side with smaller pictures taking over just 1/2 of the screen estate.

**Comment by westnordost** [comment]
My suggestion would be to display the bottom sheet as normal, but then sliding in from the right and taking up the right part of the screen only when the quest from is open. This is how Maps.Me does it.

**Comment by ENT8R** [comment]
What do you think about this:
<img src="https://user-images.githubusercontent.com/25306497/29488732-5fb04850-8511-11e7-8f17-da0d6762ab96.png" width="325px"> <img src="https://user-images.githubusercontent.com/25306497/29488733-61890b62-8511-11e7-9fee-171e21c0670b.png" width="325px">
<img src="https://user-images.githubusercontent.com/25306497/29488734-64562c8a-8511-11e7-92aa-472623923822.png" width="325px"> <img src="https://user-images.githubusercontent.com/25306497/29488735-6710ac20-8511-11e7-8200-9623f3c1087f.png" width="325px">

**Comment by rugk** [comment]
In such a condensed way only possible on tablets. (But looks nice there.) On phones the height is smaller, so you .

Only as for the [last image](https://user-images.githubusercontent.com/25306497/29488735-6710ac20-8511-11e7-8200-9623f3c1087f.png) I'd say the question (title) should not overlap the blue "StreetComplete" bar. (Although that bar is worth another issue: https://github.com/westnordost/StreetComplete/issues/507)

**Comment by westnordost** [comment]
In my opinion:
1. It should slide in from the left so it does not hide the compass and so that the X-button is at an expected position
2. It should slide in (from the screen edge) already fully expanded. It looks a bit weird with this L-shape
3. The focussed element should of course be in the center of the still visible area of the screen

Point 3 will be hard to implement because currently I am using a hack to manually offset the map centering because the bottom sheet is above the map layout (thus the map view spans still the whole screen). See https://github.com/tangrams/tangram-es/issues/1622
The proper solution would be to change the layout so that the map view really only spans the area it is visible - so when the bottom sheet comes up, the map view becomes smaller. (Basically, think vertical LinearLayout - first child is map view, second child is the bottom sheet).
I didn't do this yet because this "BottomSheet" thing is a special thing from google and I am not sure if it would work that way. If not, the behavior would need to be programmed ourselves. This is not that much work, but I haven't gotten around to do it yet.

**Comment by rugk** [comment]
I think 3. is not really necessary. I just looked at Maps.me and even they don't do it. When the info area is shown/expands there, it may actually hide the point it is about. Not nice, but on the other hand, it may also be a bit awkward if the map moves to the right when you open a quest in that way.
However, looking at how it is currently done in StreetComplete it may not be that bad, as the map currently already zooms to the quest…

**Comment by ENT8R** [comment]
> It should slide in (from the screen edge) already fully expanded

That's right... But we can only expand quests like street surface or roof shape. The other quests would look weird and have much white space... What do you think?

BTW, This is how it looks at the moment:
<img src="https://user-images.githubusercontent.com/25306497/29493866-57633336-859f-11e7-862f-c1039f8956ff.png" width="325px"> <img src="https://user-images.githubusercontent.com/25306497/29493867-597ed79c-859f-11e7-87fc-7d1e0b784116.png" width="325px">
<img src="https://user-images.githubusercontent.com/25306497/29493868-5ca51bfc-859f-11e7-92c2-dc9eae758465.png" width="325px">

**Comment by westnordost** [comment]
The X-button placement is fine this way.

Well, the question is what looks weirder. This L-shape or white space. But anyway, I won't block your PR if you prefer the L. I find point 3 though an important usability consideration and I predict that issues for that will pop up if that is not done now.
And that point 2 is done however kinda presupposes point 3 - except if you want to build upon my hack-solution with the calculated offset. I can't blame you if you did since I introduce that hack in the first place. But still, it would be nicer (and probably less code) without that hack.

**Comment by rugk** [comment]
Maybe the L shape looks better when the "sliding quest window" has some shadow below, so it looks like a popup or so.

**Comment by ENT8R** [comment]
The always fully expanded version of the layout looks like this:
<img src="https://user-images.githubusercontent.com/25306497/29495730-3b653dfc-85c5-11e7-986f-3e60812165be.png" width="325px"> <img src="https://user-images.githubusercontent.com/25306497/29495733-3d4cf5f6-85c5-11e7-9180-e09e1740988b.png" width="325px">
What do you think?

**Comment by rugk** [comment]
In your mockup the question still overlaps the "StreetComplete" title bar.

Apart from that the fully expanded version for the house number quest really looks silly, IMHO. Especially the right border is strange, as there a street is below the quest and both the backgroud color of the street and the quest are white. So there should definitively a border at the right, even if it is just a small shadow.

~~Generally saying I think on tablets you may actually not need that (slide in from left) behavior as you have enough vertical space and thus the quest window can just be centered in the middle at the bottom.~~
~~And for phones we do not have to care about L form vs "streched" as there, the quest should…~~

Just saw your screenshots are phone screenshots, so… mhh… okay.

**Comment by westnordost** [comment]
> In your mockup the question still overlaps the "StreetComplete" title bar.

This is ok / intended. It does so in the portrait version as well.

I am also indecisive about the L vs the white space, now that I see your screenshot. @rugk is certainly right that a shadow (elevation) is missing. Perhaps for forms that don't need that much space, it would look better if the content would flow in the middle.

**Comment by ENT8R** [comment]
I will try to implement this in the next days!

**Comment by rugk** [comment]
>  It does so in the portrait version as well.

Okay, then it should however be on the point, i.e. not shifted:
![grafik](https://user-images.githubusercontent.com/11966684/29497569-b80f69de-85ea-11e7-9a0b-380879026bbe.png)
…, but, on the background bar:
![grafik](https://user-images.githubusercontent.com/11966684/29497579-d5fd9cf4-85ea-11e7-9a26-6d46caa74e55.png)

So that the user sees that as one bar.

**Comment by ENT8R** [comment]
> Okay, then it should however be on the point, i.e. not shifted

I think the problem is the length of the question. If two or more rows are used for the question, the title bar automatically gets larger...

**Comment by rugk** [comment]
Oh, yeah, right. But in this case, I think, it is okay to just wrap the text and increase the height of the blue bar in the quest part, as it is usually done. That's the easiest solution…

**Comment by ENT8R** [comment]
Found a solution! 🎉 The text is now getting smaller automatically if the question gets too long!

**Comment by westnordost** [comment]
No, please don't. Readability beats "nice alignment".

**Comment by ENT8R** [comment]
> No, please don't. Readability beats "nice alignment"

That's right... I tested it right now and especially long questions like "What are the opening hours of [name of the shop]" are really hard to read! So I have to undo the changes and we should forget the title bar...

**Comment by ENT8R** [comment]
At the moment my problem is the shadow or elevation on the right side of the quest, because `android:elevation` can be used only after API 21 and the centering of the StreetSurface quest.
All other quests look like this:
![screenshot_1503331114](https://user-images.githubusercontent.com/25306497/29527859-64d7b856-869a-11e7-90f5-f9fb36f19365.png)
But the StreetSurface quest looks like this and I can't scroll up anymore...Do you have a solution?
![screenshot_1503331128](https://user-images.githubusercontent.com/25306497/29527916-7fe311a4-869a-11e7-875f-577d5dd1e6cd.png)

**Comment by westnordost** [comment]
Oh I think the centering turns out rather nicely, don't you think?

Yeah, elevation. There must be some backwards compatible solution where on api 21 and above, this is used, and below, something else.

**Comment by ENT8R** [comment]
Yes, it looks pretty nice. But I have a problem with the RoadSurface quest. Here the centering has a wrong effect... If I open a RoadSurface quest, I can't scroll up anymore (see the last picture in my last comment). If I even try to input something, the selected choice is at the top and I can't scroll up to the other answers anymore too. Do you know how to fix this?

**Comment by westnordost** [comment]
Of course I don't because I don't know how you implemented it. My first go would be to put the scroll view with possibly `android:fillViewport="true"` and `wrap_content` into a `RelativeLayout` (if it isn't yet) and set centerVertical or how this attribute was called to true.

**Comment by rugk** [comment]
Why not create a PR? Then you can both look at the source and don't have to guess the implementation.

**Comment by ENT8R** [comment]
> because I don't know how you implemented it

Oops. I didn't thought about this :smile:
But I think your idea might work, so I will test this today! And after that I will create a PR.
